src/config.cpp

```diff
@@ -153,6 +192,16 @@ void PorymapConfig::setPrettyCursors(bool enabled) {
     this->save();
 }
 
+void PorymapConfig::setGeometry(QByteArrayList geometry) {
+    this->restoreGeometry = true;
```

Comment:
This function should either take an argument for each parameter, or there should be aseparate function for each geometry that porymap saves.  Relying on a properly formatted array is more fragile and unreadable.

Reply:
You'd then have to check to make sure the geometries had non-null/zero values before setting window state on program start, rather than relying on this boolean.

Reply:
It seems that anything invalid will just restore to the default settings. I've put several random junk things in the config for these and it never messed up.